calico/felix/felix.py

```diff
@@ -166,25 +181,32 @@ def _main_greenlet(config):
             v4_ep_manager,
             v4_fip_manager,
 
-            v6_raw_updater,
-            v6_filter_updater,
-            v6_nat_updater,
-            v6_ipset_mgr,
-            v6_rules_manager,
-            v6_dispatch_chains,
-            v6_ep_manager,
-            v6_fip_manager,
-
             iface_watcher,
             etcd_api,
         ]
 
+        if v6_enabled:
+            top_level_actors += [
+                v6_raw_updater,
+                v6_filter_updater,
+                v6_nat_updater,
+                v6_ipset_mgr,
+                v6_rules_manager,
+                v6_dispatch_chains,
+                v6_ep_manager,
+                v6_fip_manager,
+            ]
+
         monitored_items = [actor.greenlet for actor in top_level_actors]
 
         # Install the global rules before we start polling for updates.
         _log.info("Installing global rules.")
-        install_global_rules(config, v4_filter_updater, v6_filter_updater,
-                             v4_nat_updater, v6_nat_updater, v6_raw_updater)
+        if v6_enabled:
+            install_global_rules(config, v4_filter_updater, v6_filter_updater,
```

Comment:
`install_global_rules()` is getting very messy.  How about changing that method to remove the loops over v4 and v6 and just passing in one set of updaters at a time.  Then, here, we'd call the method twice for v4+v6 but only once for v4-only.

The parts at the top of the method are for IPIP, which is v4-only so those could be pulled out into a second method that we call beforehand.